File: app/src/main/java/com/chaitupenju/popularmovies2/trailers/Trailer.java
package com.chaitupenju.popularmovies2.trailers;

import android.util.Log;

import java.util.ArrayList;

public class Trailer{
    String title;
    String url;
    private static final String delimitStr = "-TS-";
    private static final String delimitTrailerStr = "-TOBJ-";

    public Trailer(String title, String url) {
        this.title = title;
        this.url = url;
    }

    public static String arrayToString(ArrayList<Trailer> trailers){
        StringBuilder res = new StringBuilder();
        try {
            for (int i = 0; i < trailers.size(); i++) {
                res.append(trailers.get(i).title).append(delimitTrailerStr).append(trailers.get(i).url);
                if (i < trailers.size() - 1) {
                    res.append(delimitStr);
                }
            }
        }
        catch(NullPointerException e){
            return "";
        }
        return res.toString();
    }

    public static ArrayList<Trailer> stringToArray(String string){
        String[] elements = string.split(delimitStr);
        ArrayList<Trailer> res = new ArrayList<>();
        for (String element : elements) {
                String[] item = element.split(delimitTrailerStr);
            try{
                res.add(new Trailer(item[0], item[1]));
            }catch (IndexOutOfBoundsException e) {
                e.printStackTrace();
            }
        }
        Log.i("STR2ARR",res.toString());
        return res;
    }

}

File: app/src/main/java/com/chaitupenju/popularmovies2/models/Trailer.kt
package com.chaitupenju.popularmovies2.models

import android.util.Log

data class Trailer(
    val title: String,
    val url: String
) {

    companion object {
        private const val delimitStr = "-TS-"
        private const val delimitTrailerStr = "-TOBJ-"

        fun arrayToString(trailers: ArrayList<Trailer>): String {
            val res = StringBuilder()

            try {
                for (i in trailers.indices) {
                    res.append(trailers[i].title).append(delimitTrailerStr).append(
                        trailers[i].url
                    )
                    if (i < trailers.size - 1) {
                        res.append(delimitStr)
                    }
                }
            } catch (e: NullPointerException) {
                return ""
            }

            return res.toString()
        }

        fun stringToArray(string: String): ArrayList<Trailer> {
            val elements = string.split(delimitStr).toTypedArray()
            val res = ArrayList<Trailer>()

            for (element in elements) {
                val item = element.split(delimitTrailerStr).toTypedArray()
                try {
                    res.add(Trailer(item[0], item[1]))
                } catch (e: IndexOutOfBoundsException) {
                    e.printStackTrace()
                }
            }

            Log.i("STR2ARR", res.toString())

            return res
        }
    }
}